parser-typechecker/src/Unison/Typechecker/Context.hs

```diff
@@ -544,6 +544,23 @@ debugShow :: (Show a) => a -> Bool
 debugShow e | debugEnabled = traceShow e False
 debugShow _ = False
 
+debugTrace :: String -> Bool
+debugTrace e | debugEnabled = trace e False
+debugTrace _ = False
+
+showType :: Var v => Type.Type v a -> String
+showType ty = TP.prettyStr (Just 120) PPE.empty ty
+
+debugType :: Var v => String -> Type.Type v a -> Bool
+debugType tag ty
+  | debugEnabled = debugTrace $ "(" <> show tag <> "," <> showType ty <> ")"
+  | otherwise = False
+
+debugTypes :: Var v => String -> Type.Type v a -> Type.Type v a -> Bool
+debugTypes tag t1 t2
+  | debugEnabled = debugTrace $ "(" <> show tag <> ",\n  " <> showType t1 <> ",\n  " <> showType t2 <> ")"
+  | otherwise = False
```

Comment:
@dolio these might be handy next time you're debugging something. Also, `TypePrinter.prettyStr` which is used here does no normalization, so it just prettily prints exactly what's there.

Reply:
Nice. That does sound handy.

Reply:
cc @unisonweb/ucm